### ISSUE TITLE
CMS.py Contains TWO Conflicting Definitions of RecipientKeyIdentifier

### ISSUE THREAD

**Issue by cjamescook** [wbond/asn1crypto#243]
cms.py contains TWO definitions of "class RecipientKeyIdentifier". Unfortunately, the first defines the field "subject_key_identifier" and the latter defines the field "subjectKeyIdentifier". This snowballs, causing problems due to Python binding time rules. That is, the definition of KeyAgreementRecipientIdentifier picks up the former definition, but a program that imports cms.py will pick up the latter.

To wit, the following fails:

```
>>> temp=RecipientKeyIdentifier()
>>> temp['subjectKeyIdentifier'] = b'12345678'
>>> KeyAgreementRecipientIdentifier(name='r_key_id', value=temp)
```

The following works, but it took me over a day to figure out what was going on:

```
>>> temp.native
OrderedDict([('subjectKeyIdentifier', b'12345678'), ('date', None), ('other', None)])
>>> temp2=util.OrderedDict([('subject_key_identifier', b'12345678'), ('date', None), ('other', None)])
>>> KeyAgreementRecipientIdentifier(name='r_key_id', value=temp2)
```

**Comment by geitda** [comment]
Commit https://github.com/wbond/asn1crypto/commit/f003c67831782ed8c225c8b11c3aa3d0cc56fb1d erroneously duplicated the `RecipientKeyIdentifier` class, which effectively changed the field name from underscores to camelCase, and gums things up as @cjamescook noted. Deleting the duplicate and sticking to using `'subject_key_identifier'` seems to make any issues go away, but this might break any other code written since 2019 that interfaces that class with the `'subjectKeyIdentifier'` name.
Looks like there aren't any tests to touch `RecipientKeyIdentifier`, `KeyAgreementRecipientIdentifier`, or `SMIMEEncryptionKeyPreference` so that's why this fell through the cracks.
I may submit a PR that removes the duplicate and adds some tests if there's community interest for it and I get enough free time.

**Comment by wbond** [comment]
I'll have to add some code to work-around the new name.

**Comment by wbond** [comment]
I have a PR at #266 that should work around this issue.

@cjamescook @geitda Do either of you have an interest in helping provide some sort of example file that we could use a test fixture for this structure?

Alternatively @commonism since you originally added this code, do you have something we could use for a test?

**Comment by commonism** [comment]
Hi,

IIRC the structures were required to parse SMIME signatures from Outlook signed mails.
I never had to access the fields - I do not have code which will change in behavior when the field gets renamed.
I did not have outlook, could not include foreign signatures and failed to create them manually, so there is no tests on this.

Maybe consider using https://pre-commit.ci/ to make use of something like black/pylint/flake8… to detect re-definition/* and block merging. pre-commit is great.

**Comment by commonism** [comment]
Possible to create something from 
https://learn.microsoft.com/en-us/openspecs/exchange_server_protocols/ms-oxoabk/2653a83f-ace5-4543-8908-d6557e2f6ed6
https://learn.microsoft.com/en-us/openspecs/exchange_server_protocols/ms-oxoabk/5cac7bfa-cb9e-42a6-9d22-84a3b6f155d3
?

**Comment by wbond** [comment]
This has been fixed as of 8609892a88f571bc10110603c173832cd100cb44 and will be part of the next release
